docs_src/design/adr/device-service/0011-DeviceService-Rest-API.md

```diff
@@ -144,9 +157,13 @@ ie, `new-value = (current-value & !mask) | request-value`
 
 The combination of mask and shift can therefore be used to access data contained in a subdivision of an octet.
 
+It is possible that following the application of the specified transformations, a value may exceed the range that may be represented by its type. Should this occur on a set operation, a suitable error should be logged and returned, along with the `Bad Request` http code 400. If it occurs as part of a get operation, the Reading's value should be set to the String `"overflow"` and its valueType to `String`.
+
 #### Assertions and Mappings 
 
-Assertions are another attribute in a device resource's PropertyValue which specify a string value which the result is compared against. If the comparison fails, then the result is set to a string of the form *"Assertion failed for device resource: \<name>, with value: \<result>"*, this also has a side-effect of setting the device operatingstate to `DISABLED`. A 500 status code is also returned.
+Assertions are another attribute in a device resource's PropertyValue, which specify a string which the reading value is compared against. If the comparison fails, then the http request returns a string of the form *"Assertion failed for device resource: \<name>, with value: \<value>"*, this also has a side-effect of setting the device operatingstate to `DISABLED`. A 500 status code is also returned. Note that the error response and status code should be returned regardless of the `ds-returnevent` setting.
+
+Assertions are also checked where an event is being generated due to an AutoEvent, or asynchronous readings are pushed. In these cases if the assertion is triggered, an error should be logged and the operating state should be set as above.
 
 Mappings may be defined in a deviceCommand. These allow Readings of string type to be remapped. Mappings are applied after assertions are checked, and are the final transformation before Readings are created. Mappings are also applied, but in reverse, to settings (`PUT` request data).
```

Comment:
I know this isn't a change in this PR, but it's slightly confusing as currently worded with respect to settings. For readings, all of the transforms are applied in order, then assertions are checked, and finally mappings are applied.

For settings I'm assuming mappings are applied first, then all of the transforms are applied in reverse order **then** assertions are checked?